### ISSUE TITLE
Mustache-notation is not interpreted before sending userTasks

### ISSUE THREAD

**Issue by heikomat** [process-engine/process_engine#13]
#### Describe your issue

When receiving userTasks via messageBus or `/user_task_data/:user_task_id`-route, notations like these are interpreted [here](https://github.com/process-engine/process_engine/blob/3ca4f46b4b5280a5d6bc7eb10dedcb3b6024b273/src/entity_types/node_instance.ts#L657):

```
${ "message": "Der Preis ist " + current + " " + history.ut_WaehleKlasse.currency, "layout": [ { "key": "confirm", "label": "Bestätigen" }, { "key": "decline", "label": "Verwerfen", "isCancel": true } ]};
```

While mustache-notations like these are not interpreted at all:
```
${ "message": "Der Preis ist {{{current}}} {{{history.ut_WaehleKlasse.currency}}}", "layout": [ { "key": "confirm", "label": "Bestätigen" }, { "key": "decline", "label": "Verwerfen", "isCancel": true } ]}
```

before charon started using the consumer-client, charon itself had mustache as dependency, and interpreted these strings itself. The ConsumerClient however, doesn't handle mustache-strings, therefore they are not displayed correct in charon

<img width="759" alt="bildschirmfoto 2017-10-31 um 23 29 39" src="https://user-images.githubusercontent.com/20770029/32251968-99e0bfbc-be93-11e7-8712-a8b969f45e10.png">

 This issue is a result from the conversation in https://github.com/process-engine/charon/issues/41

#### How can others reproduce the issue?

Just follow the regular skeleton-reservation-demo setup and start the reservation-process

#### Possible solution

There are three possible options.

- We can make the backend interpret mustache-strings before returning the userTask
- We can drop the support for mustache-strings
- We can make the consumerClient interpret the mustache-strings (I really don't recommend that option).

**_IF_** we decide to keep the support for mustache-strings in extensions, i'm very much for interpreting them in the backend (which is why i opened this issue here), because why would we handle them differently than the Function-notation? Also interpreting in the backend is easier to implement (the method in the backend is already named `parseExtensionProperty`, and would just need a couple extra lines).

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

**Comment by heikomat** [comment]
@ppoetschke @zimoony @Paulomart @5ebastianMeier If no one has an opinion on which solution to choose, i will (in a couple of days, let's say friday) either implement the support for mustache-strings in the backend, or declare them as not supported.

If however someone has an opinion on that, please let me know

**Comment by ppoetschke** [comment]
This kind of functionality should be neither part of the backend nor of the consumer client api. It's the sole task of the specific widget to interpret template strings.
The backend is already evaluating expressions and the correct way to render strings backend side is concatenating them appropriately.
The purpose of the consumer client api module is acting like a proxy between backend and the implementation of the consumer (in this case the ui layer). It should not modify the config or anticipate rendering. But it can pass the data object used for interpreting a template. Leaving the widget handle templates gives also more flexibility as every control character supported by a widget can be used in user task config.

**Comment by heikomat** [comment]
@ppoetschke so in short, you are for declaring it obsolete.

That's fine with me. However, i have a slightly different understanding regarding the purpose of the consumer_client. While it should definitely not expect any specific framework or environment, it may very well anticipate that userTasks are probably going to be displayed in one way or another. That is the reason why it introduces "userTaskConfigs", which are essentially userTask-Objects, but interpreted in the backend and rearranged in a way that makes displaying a userTask easier compared to userTaskEntities from the dataStore.

The consumer_client is a convinience-api. "being a proxy" is in itself not a useful purpose for a convinience-api. The purpose should be to make implementing a client easier and it does this by defining what a consumer can and can't do. These things a consumer can and can't do are defined by the methods and events it provides to the person using it. These methods happen to abstract away the process-engine-communication in order to better communicate that goal/purpose to the person using it.

in short: There is a reason the user asks the consumer_client for userTasks, instead of asking the process-engine for some datastore-data. To achieve this, the process-engine-communcation needs to be abstracted. This doesn't make "being a proxy" the purpose of the consumer_client, it's just a way for the consumer-client to fullfill its purpose.

This is at least how i see it, but this is probably a discussion better held somewhere else (aka. not this issue)

**Comment by ppoetschke** [comment]
@heikomat No, I‘m not declaring it obsolete, it should be implemented at the right place.
Concerning your opinion about the purpose of the consumer client, it‘s not contradictionary to mine, you explained it more in detail.

**Comment by Paulomart** [comment]
As I already noted [here](https://github.com/process-engine/charon/issues/41#issuecomment-339950897), I dont see a need for the mustage notation. @ppoetschke could you explain the different use cases, I may miss?

**Comment by 5ebastianMeier** [comment]
Placeholder resolution is part of the backend process imho - it's actually something the client must not do - it could only provide additional data for the placeholders during the request


the API should also support a language header in a later iteration and use that to reply with a localized and fully resolved response

**Comment by ppoetschke** [comment]
Please lets move this to a conceptual document. Legacy BPMNs should be updated by removing {{}}  .

**Comment by heikomat** [comment]
@ppoetschke so you **are** declaring it obsolete

**Comment by ppoetschke** [comment]
@heikomat No, I 'm saying we need more discussion to implement it correctly for the price to break existing processes temporarily.

**Comment by Paulomart** [comment]
For me it looks like there are currently to ways to express the placeholder functionality. We should drop one of them and only support one.
